src/highlight-component.js

```diff
@@ -0,0 +1,60 @@
+function HighlightComponent() {
+  this.state = false;
+  this.components = extractComponentsFromPage();
+
+  function extractComponentsFromPage() {
+    return $('[class*="app-c"], [class*="pub-c"], [class*="govuk"]')
+      .toArray()
+      .reduce(function(array, element) {
+        var blockRegex = /(app-c-|pub-c-|govuk-c-|govuk-)([^ _\n]*(?=[ \n]|$))/;
+        var match = $(element).attr('class').match(blockRegex);
+
+        if (match) {
+          array.push({
+            element: element,
+            prefix: match[1],
+            name: match[2]
+          });
+        }
+
+        return array
+      }, []);
+  }
+
+  this.components.forEach(setupComponent.bind(this));
+
+  function setupComponent(component) {
+    var $element = $(component.element);
+    $element.attr('data-component-name', component.name);
+
+    $element.on('click', function() {
+      if (this.state) {
+        window.open( Helpers.documentationUrl(component) );
+      }
+    }.bind(this));
+  }
+
+  chrome.runtime.onMessage.addListener(function (request) {
+    if (request.trigger == 'toggleState') {
+      this.toggleState();
+    }
+  }.bind(this));
+}
+
+HighlightComponent.prototype.toggleState = function () {
+  this.state = !this.state;
+
+  for (var i = 0; i < this.components.length; i++) {
+    $(this.components[i].element).toggleClass('highlight-component', this.state);
+  }
+}
+
+var Helpers = {
+  documentationUrl: function (component) {
+    if (component.prefix.startsWith('app-c')) {
+      return "https://government-frontend.herokuapp.com/component-guide/" + component.name
```

Comment:
This won't always be true:
eg https://github.com/alphagov/finder-frontend/pull/312 and https://finder-frontend.herokuapp.com/component-guide (when it's merged)

Reply:
The rendering app should already be determined by the addon. But app names aren't always going to be a 1:1 relationship with their Heroku app name 🤔

Reply:
Perhaps we can put a link to the guide on the app documentation page (like https://docs.publishing.service.gov.uk/apps/government-frontend.html) and link to that?

Reply:
But that wouldn't link directly to the component.

Reply:
Haha, I didn't scroll to the right!